packages/query-core/src/queryClient.ts

```diff
@@ -204,7 +210,11 @@ export class QueryClient {
   getQueryState<TQueryFnData = unknown, TError = DefaultError>(
     queryKey: QueryKey,
   ): QueryState<TQueryFnData, TError> | undefined {
-    return this.#queryCache.find<TQueryFnData, TError>({ queryKey })?.state
+    const queryHash = hashQueryKeyByOptions(
+      queryKey,
+      this.#defaultOptions.queries,
```

Comment:
same here, we really shouldn't be using only the default options. Maybe it should be a function on the QueryCache itself ?

Reply:
Ah, thank you. I wasn't clear on the difference between `defaultOptions` and `queryDefaults`. 

The updated code is compact enough that a separate method isn't necessary IMO. 

On a side-note: I'm a bit surprised that TS lets me return `unknown` as `TInferredQueryFnData` without an explicit cast.

Reply:
> On a side-note: I'm a bit surprised that TS lets me return unknown as TInferredQueryFnData without an explicit cast.

yeah interesting - but it was the same with the previous implementation - `.find` also returns `unknown`